spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -104,8 +101,16 @@ else if (tokens.length != 1) {
 					channel = new DirectChannel();
 					this.beanFactory.registerSingleton(beanName, channel);
 					channel = (MessageChannel) this.beanFactory.initializeBean(channel, beanName);
-					Binder<MessageChannel> binder = binderFactory.getBinder(transport);
-					this.dynamicDestinationsBindable.addOutputBinding(beanName, binder.bindProducer(destinationName, channel, producerProperties));
+					@SuppressWarnings("unchecked")
+					Binder<MessageChannel, ?, ProducerProperties> binder =
+							(Binder<MessageChannel, ?, ProducerProperties>) binderFactory.getBinder(binderName);
+					// TODO: need the props to return some defaults if not found
```

Comment:
would be a good idea to create a GH issue for this TODO if it won't be address in the scope of this PR